pip-8210

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+if [[ ${-} != *i* ]]; then
+  source_once &> /dev/null && return 0
+fi
+
+#*# pip-8210
+
+#**
+# .. file:: pip-8210
+#
+# This script allows us to install a package outside of pip-tools. In order to accomplish correctly, we use the ``requirements.txt`` file as a constraint files,and this and handle corner cases like pip issue 8210.
+#
+# It add the -c constraints to ``pip install``, ``pip download``, and ``pip wheel``
+#
+# You should activate your virtualenv before calling this; it will use the ``pip`` on the path.
+#
+# :Arguments: * [--if-not-found <executable name>] - Only runs the ``pip`` command if a specified executable is not found.
+#             * ``$1``... - Args to pass to ``pip``. E.g. ``install pip-tools``
+#
+# .. note::
+#
+#    If the first package is ``pip-tools``, it will install pip-tools even if it is not in ``requirements.txt`` (necessary to bootstrap new ``requirements.txt`` files). If additional arguments were added in this corner case, the return value will be non-zero since not all the packages got installed, but ``pip-tools`` did get installed.
+#
+# .. seealso::
+#
+#    https://github.com/pypa/pip/issues/8210
+#
+# .. function:: pip-8210
+#
+# Same syntax as :file:`pip-8210`
+#**
+
+if [ -z "${VSI_COMMON_DIR+set}" ]; then
+  VSI_COMMON_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.."; pwd)"
+fi
+
+if [ -f "${VSI_COMMON_DIR}/linux/command_tools.bsh" ]; then
+  source "${VSI_COMMON_DIR}/linux/command_tools.bsh"
+else
+  # function parse-generic(){ :;}
+  function parse-generic()
+  {
+    command_args=()
+    subcommand_args=()
+
+    while (( ${#} )); do
+      case "${1}" in
+        -*)
+          # Anything that takes one arguments from the array arguments_with_value
```

Comment:
```suggestion
          # Anything that takes one argument from the array arguments_with_value
```

Reply:
Shouldn't arguments_with_value be documented as an input of some sort

Reply:
This is just an exact copy from `command_tools.bsh` (for portability).